src/api/PDFPage.ts

```diff
@@ -566,6 +571,73 @@ export default class PDFPage {
     this.node.wrapContentStreams(startRef, endRef);
   }
 
+  /**
+   * Scale the content of a page. This is useful after resizing an exisiting page.
+   * This scales only the content not the annots. See also: [[scaleAnnots]]
+   * ```js
+   * // bisect the size of the page
+   * p.setSize(p.getWidth() / 2, p.getHeight() / 2);
+   *
+   * // scale the content of the page down by 50% in x and y
+   * page.scaleContent(0.5, 0.5);
+   * ```
+   * @param x The factor by wich the x-axis for the content should be scaled (e.g. 0.5 is 50%)
+   * @param y The factor by wich the y-axis for the content should be scaled (e.g. 0.5 is 50%)
+   */
+  scaleContent(x: number, y: number): void {
+    assertIs(x, 'x', ['number']);
+    assertIs(y, 'y', ['number']);
+
+    this.node.normalize();
+    this.getContentStream();
+
+    const start = this.createContentStream(pushGraphicsState(), scale(x, y));
+    const startRef = this.doc.context.register(start);
+
+    const end = this.createContentStream(popGraphicsState());
+    const endRef = this.doc.context.register(end);
+
+    this.node.wrapContentStreams(startRef, endRef);
+  }
+
+  /**
+   * Scale the annots of a page. This is useful if you want to scale a page with comments or other annots.
+   * ```JS
+   * // scale the content of the page down by 50% in x and y
+   * page.scaleContent(0.5, 0.5);
+   *
+   * // scale the content of the page down by 50% in x and y
+   * page.scaleAnnots(0.5, 0.5);
+   * ```
+   * See alos: [[scaleContent]]
+   * @param x The factor by wich the x-axis for the annots should be scaled (e.g. 0.5 is 50%)
+   * @param y The factor by wich the y-axis for the annots should be scaled (e.g. 0.5 is 50%)
+   */
+  scaleAnnots(x: number, y: number) {
+    const annots = this.node.Annots();
+
+    // loop annotations
+    annots?.asArray().forEach((ref: PDFObject) => {
+      const i = annots.indexOf(ref);
+
+      if (i === undefined) return;
+
+      const annot = annots.lookup(i, PDFDict);
+
+      if (!annot) return;
+
+      ['RD', 'CL', 'Vertices', 'QuadPoints', 'L', 'Rect'].forEach((el) => {
+        const list = annot.get(PDFName.of(el)) as PDFArray;
+        if (list) this.scalePDFNumbers(list, x, y);
+      });
+
+      const pdfNameInkList = annot.get(PDFName.of('InkList')) as PDFArray;
+      pdfNameInkList
+        ?.asArray()
+        .forEach((arr) => this.scalePDFNumbers(arr as PDFArray, x, y));
```

Comment:
I need to improve the contributor docs on this, but:
* Please avoid using `asArray()` internally. It's just meant to be a convenience method for library consumers.
* Please also avoid using `.forEach()` and use a for-each-index loop instead. If you look through the codebase you'll see that's the pattern currently in use.